### DISCUSSION TITLE
Can`t correctly use renovate-config from repository

### DISCUSSION THREAD

**Discussion by pshakhov** [renovatebot/renovate#10357] [Request Help] [answered]
Hi all.

Im using this renovate.json config:
```typescript
{
  "name": "@top-renovate/renovate-config",
  "author": "Renovate Bot <company-service@company.ru>",
  "description": [ "RenovateBot config presets for BPM" ],
  "draftPR": true,
  "maintainers": [
    "Pavel Shakhov <p.shakhov@company.ru>"
  ],
  "keywords": [
    "Renovate",
    "Bot",
    "Presets",
    "TOP"
  ],
  "renovate-config": {
    "base-backend-kotlin": {
      "groupName": "all",
      "description": "Default base configuration for TOP Kotlin backend applications",
      "extends": [
        ":separateMajorReleases",
        ":combinePatchMinorReleases",
        ":ignoreUnstable",
        ":prImmediately",
        ":semanticCommits",
        ":semanticCommitTypeAll(renovatebot)",
        ":updateNotScheduled",
        ":automergeDisabled",
        ":ignoreModulesAndTests"
      ],
      "baseBranches": [
        "develop",
        "master",
        "test"
      ],
      "packageRules": [
        {
          "groupName": "all",
          "packagePatterns": [
            "org.jetbrains.kotlin"
          ]
        },
        {
          "groupName": "all",
          "packagePatterns": [
            "org.jetbrains.teamcity:"
          ]
        },
        {
          "groupName": "all",
          "packagePatterns": [
            "io.sentry"
          ],
          "allowedVersions": "4.3.0"
        },
        {
          "groupName": "spring",
          "packagePatterns": [
            "org.springframework:",
            "org.springframework.boot",
            "org.springframework.cloud"
          ]
        },
        {
          "groupName": "all",
          "packagePatterns": [
            "org.junit.vintage",
            "org.junit.platform",
            "org.junit.jupiter"
          ]
        },
        {
          "groupName": "all",
          "packagePatterns": [
            "org.mockito:",
            "com.nhaarman:"
          ]
        },
        {
          "groupName": "all",
          "packagePatterns": [
            "io.springfox:",
            "org.hidetake.swagger"
          ]
        },
        {
          "groupName": "all",
          "packagePatterns": [
            "io.gitlab.arturbosch.detekt"
          ]
        },
        {
          "groupName": "all",
          "packagePatterns": [
            "org.liquibase"
          ]
        },
        {
          "groupName": "all",
          "packagePatterns": [
            "org.sonarqube"
          ]
        },
        {
          "groupName": "all",
          "packagePatterns": [
            "io.springfox"
          ]
        }
      ]
    }
  }
}
```

placed at separate repo:
https://gitlab.company.ru/top-infra/renovate-config.git

I try to use it by:
```typescript
module.exports = {
    endpoint: 'https://gitlab.company.ru/api/v4/',
    token: 'toast',
    printConfig: true,
    gitAuthor: 'Renovate Bot <company-service@company.ru>',
    platform: 'gitlab',
    logLevel: 'debug',
    // logFile: 'renovate.log',
    logFileLevel: 'debug',
    onboarding: true,
    onboardingConfig: {
        extends: [
            "local>top-infra/renovate-config:base-backend-kotlin"
        ],
    },
    prFooter: 'This PR has been generated by [Renovate Bot](https://gitlab.company.ru/top-infra/top-renovate-bot).',
    pruneStaleBranches: false,
    repositories: [
        'top-infra/toast'
    ]
};
```
in config.js in bot repo:
https://gitlab.company.ru/top-infra/top-renovate-bot.git

In logs see my custom config, that contains custom parameters (and description on screenshot).
But seems renovate uses the undefined (not my custom and not default) config - at least MRs is separated for each dependency, not only by major|minor.
![image](https://user-images.githubusercontent.com/32340504/121243957-fece6e80-c8a6-11eb-9c67-c40500d03deb.png)

Please help, what i do wrong.

**Comment by viceice** [comment]
Please format your code

**Reply by pshakhov** [reply]
> Please format your code

Im new here, thanks, i did it.

**Reply by HonkingGoose** [reply]
@pshakhov Another cool thing: you can use `javascript` or `typescript` right after the 3 backticks (```) to get syntax highlighting.
This also works for things like JSON/YAML output, and for many other programming languages.

````
```typescript
Code goes here
```
````

<details><summary>Click me to see a snippet of your code with syntax highlighting</summary>

```javascript
module.exports = {
    endpoint: 'https://gitlab.company.ru/api/v4/',
    token: 'toast',
    printConfig: true,
    gitAuthor: 'Renovate Bot <company-service@company.ru>',
    platform: 'gitlab',
    logLevel: 'debug',
    // logFile: 'renovate.log',
    logFileLevel: 'debug',
    onboarding: true,
    onboardingConfig: {
        extends: [
            "local>top-infra/renovate-config:base-backend-kotlin"
        ],
    },
    prFooter: 'This PR has been generated by [Renovate Bot](https://gitlab.company.ru/top-infra/top-renovate-bot).',
    pruneStaleBranches: false,
    repositories: [
        'top-infra/toast'
    ]
};
```

</details>

Read the [GitHub docs, creating and highlighting code blocks, syntax highlighting](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#syntax-highlighting) for more information.

**Comment by rarkins** [comment, marked as answer]
Remove the "renovate-config" wrapper object from your renovate.json and love everything inside up to the top level

**Reply by pshakhov** [reply]
@rarkins , thanks, i did it, but didnt work as expected. 

Seems i still dont understand how to use the structure of config when applying and configure it.
How i must search the preset in repository with preset in group of our team?

The group is top-infra, repo is renovate config, i tried some ways like
```"local>top-infra/renovate-config"```
```"local>top-infra/renovate-config:base-backend-kotlin"```

Try it with gitlab instead local, or change "name"? In logs i still see the config is found but it dont applied:
![image](https://user-images.githubusercontent.com/32340504/121305496-52bc7000-c906-11eb-95ab-6ce84d1e6e64.png)

**Reply by rarkins** [reply]
You have the incorrect syntax for `renovate.json` files. Almost none of this is valid:

![image](https://user-images.githubusercontent.com/6311784/121305825-58ae5300-c8fe-11eb-9938-46d821f7e82c.png)

`name` is not valid, `maintainer` is not valid, etc. And anything inside `renovate-config` is ignored because it's not a defined field either. We have a JSON schema definition for `renovate.json` which I suggest you can use, as well as a `renovate-config-validator` binary which ships with Renovate too.

Here is an example of the main preset we use: https://github.com/renovatebot/.github/blob/main/default.json

**Reply by pshakhov** [reply]
The created onboarding `renovate.json` looks this:
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "local>top-infra/renovate-config"
  ]
}
```

**Reply by pshakhov** [reply]
> You have the incorrect syntax for `renovate.json` files. Almost none of this is valid:

I copied the config from other our company project (and edited it for my purposes), that works correctly (sad but true). 
But original config publishing in Nexus, this is difference (we dont want use this way), and uses in extend like this:
```@bpm-renovate/renovate-config```
```@bpm-renovate/renovate-config:base-backend-kotlin```
thanks a lot, Ill try to fix the config and message you after.

**Reply by rarkins** [reply]
Yes, that could have worked if it was published as an npm module, but if you're using as a repo-based preset then you need to take the content inside `renovate-config` and make it at the top level of the JSON. `renovate-config-validator` should be helpful here.

**Reply by pshakhov** [reply]
Now its work, thank you a lot! Im wasted to it half of yesterday. Need to fix my example in our company wiki to avoid others time wasting.
